analysis.py

```diff
@@ -3,22 +3,22 @@
 from re import split, sub
 from typing import Optional
 
-from pandas import Series
+from pandas import DataFrame, Series
 import pandas as pd
 
 
-def categories(
+def _category_list_per_answer(
```

Comment:
Previously the summary from `categories` was missing some of the `categories` from `categories_per_answer`. Using the same code should ensure the same set of categories are kept

Reply:
Great 🙌.

The reason I didn't do this before trying to get some results out was,

'categories' dropped rows from the series to make sure the '%' column counted only those answers which had not been dropped. _I.e._ percentage of valid answers which choose Python. 'categories_per_answer' instead replaced rows with an empty string so that the return was still of the same length as the input (and row numbers match up).

Reply:
Good point, could be argued both ways. How about we return both
- % of non-empty/non-dropped responses
- % of total responses

?

Reply:
I've pushed two more commits